scapy/layers/tls/automaton_cli.py

```diff
@@ -1163,73 +1223,20 @@ def tls13_should_add_ClientHello_Retry(self):
-
-        if s.tls13_psk_secret:
-            if self.tls13_psk_mode == "psk_dhe_ke":
-                ext += TLS_Ext_PSKKeyExchangeModes(kxmodes="psk_dhe_ke"),
-                ext += TLS_Ext_SupportedGroups(groups=[_tls_named_groups[selected_group]])  # noqa: E501
-                ext += TLS_Ext_KeyShare_CH(client_shares=[KeyShareEntry(group=selected_group)])  # noqa: E501
-            else:
-                ext += TLS_Ext_PSKKeyExchangeModes(kxmodes="psk_ke")
-
-            if s.client_session_ticket:
-
-                # XXX Retrieve parameters from first ClientHello...
-                cs_cls = _tls_cipher_suites_cls[s.tls13_ticket_ciphersuite]
-                hkdf = TLS13_HKDF(cs_cls.hash_alg.name.lower())
-                hash_len = hkdf.hash.digest_size
-
-                # We compute the client's view of the age of the ticket (ie
-                # the time since the receipt of the ticket) in ms
-                agems = int((time.time() - s.client_ticket_age) * 1000)
-
-                # Then we compute the obfuscated version of the ticket age by
-                # adding the "ticket_age_add" value included in the ticket
-                # (modulo 2^32)
-                obfuscated_age = ((agems + s.client_session_ticket_age_add) &
-                                  0xffffffff)
-
-                psk_id = PSKIdentity(identity=s.client_session_ticket,
-                                     obfuscated_ticket_age=obfuscated_age)
-
-                psk_binder_entry = PSKBinderEntry(binder_len=hash_len,
-                                                  binder=b"\x00" * hash_len)
-
-                ext += TLS_Ext_PreSharedKey_CH(identities=[psk_id],
-                                               binders=[psk_binder_entry])
-            else:
-                hkdf = TLS13_HKDF("sha256")
-                hash_len = hkdf.hash.digest_size
-                psk_id = PSKIdentity(identity='Client_identity')
-                psk_binder_entry = PSKBinderEntry(binder_len=hash_len,
-                                                  binder=b"\x00" * hash_len)
-
-                ext += TLS_Ext_PreSharedKey_CH(identities=[psk_id],
-                                               binders=[psk_binder_entry])
-
-        else:
-            ext += TLS_Ext_SupportedGroups(groups=[_tls_named_groups[selected_group]])  # noqa: E501
-            ext += TLS_Ext_KeyShare_CH(client_shares=[KeyShareEntry(group=selected_group)])  # noqa: E501
-            ext += TLS_Ext_SignatureAlgorithms(sig_algs=["sha256+rsaepss"])
+        if not self.supported_groups or not self.advertised_tls_version:
+            self.vprint("No common groups found in TLS 1.3 Hello Retry Request!")
+            raise self.CLOSE_NOTIFY()
 
-        p = TLS13ClientHello(ciphers=ciphersuite, ext=ext)
-        self.add_msg(p)
-        raise self.TLS13_ADDED_CLIENTHELLO()
+        self.tls13_should_add_ClientHello()
```

Comment:
(Note to self: double check that this is iso. Common cipher checking looks wrong)

Reply:
Fixed. There was indeed a bug. Added some tests to check we don't break HelloRetryRequest.